pkgs/development/python-modules/gpt4all-bindings/default.nix

```
@@ -0,0 +1,77 @@
+{
+  lib,
+  gpt4all,
+  gpt4all-cuda,
+  buildPythonPackage,
+  stdenv,
+  requests,
+  tqdm,
+  pythonOlder,
+  setuptools,
+  wheel,
+  python,
+  cuda ? false,
+}:
+let
+  basepkg = if cuda then gpt4all-cuda else gpt4all;
+
+  gpt4all-backend = stdenv.mkDerivation {
+    pname = "gpt4all-backend";
+    inherit (basepkg)
+      src
+      version
+      cmakeFlags
+      nativeBuildInputs
+      buildInputs
+      ;
+
+    sourceRoot = "${basepkg.src.name}/gpt4all-backend";
+    installPhase = ''
+      cp -r ./ $out
+    '';
+  };
+in
+buildPythonPackage {
+  pname = "gpt4all-bindings";
```

```suggestion
  pname = "gpt4all";
```
> Attribute names in python-packages.nix as well as pnames should match the library's name on PyPI, but be normalized according to [PEP 0503](https://www.python.org/dev/peps/pep-0503/#normalized-names). This means that characters should be converted to lowercase and . and _ should be replaced by a single - (foo-bar-baz instead of Foo__Bar.baz).

https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#contributing-guidelines-contributing-guidelines